framework/src/media/InputDataSource.cpp

```diff
@@ -53,29 +53,35 @@ InputDataSource::~InputDataSource()
 {
 }
 
-void InputDataSource::registerDecoder(audio_type_t audioType, unsigned int channels, unsigned int sampleRate)
+bool InputDataSource::registerDecoder(audio_type_t audioType, unsigned int channels, unsigned int sampleRate)
 {
 	std::shared_ptr<Decoder> decoder = nullptr;
 	switch (audioType) {
 	case AUDIO_TYPE_MP3:
 	case AUDIO_TYPE_AAC:
 	case AUDIO_TYPE_OPUS:
-		decoder = std::make_shared<Decoder>(channels, sampleRate);
+		decoder = Decoder::create(getAudioType(), getChannels(), getSampleRate());
```

Comment:
getAudioType() returns mAudioType, The 'mAudioType' can be different from 'audioType' variable in this scope. Is it okay?